deploy-docker/src/index.ts

```diff
@@ -0,0 +1,100 @@
+import { promises as fs } from 'fs'
+import { getInput, setFailed } from '@actions/core'
+import { getOctokit } from '@actions/github'
+import { getBooleanInput } from '../../lib/core'
+import { getShaFromRef, getToken } from '../../lib/github'
+import { downloadS3object, getAWSCredentials } from '../../lib/s3'
+import { exec } from '@actions/exec'
+
+async function run() {
+  const component = getInput('component', { required: true })
+  const dockerUser = getInput('docker-user', { required: true })
+  const dockerPassword = getInput('docker-password', { required: true })
+  const dryRun = getBooleanInput('dry-run', false)
+  const privateKey = getInput('private-key')
+  const dockerfile = getInput('dockerfile') || './Dockerfile'
+  const ghToken = getToken()
+
+  const [awsKeyId, awsSecretKey] = getAWSCredentials()
+
+  const imageVersion = (
+    process.env['GITHUB_REF']?.slice(0, 10) == 'refs/tags/'
+      ? process.env['GITHUB_REF']
+      : process.env['GITHUB_REF']
+  )
+    ?.split('/')
+    .reverse()[0]
+  console.log(`Image version will be: ${imageVersion}`)
+  const compVersion =
+    process.env['GITHUB_REF']?.slice(0, 10) == 'refs/tags/'
+      ? process.env['GITHUB_REF']?.slice(10)
+      : process.env['GIHUB_SHA']
+  console.log(`Component version will be: ${compVersion}`)
+
+  console.log('Writing metadata to metadata.json')
```

Comment:
please make this a const at the top of the file (e.g. `const METADATA_FILENAME = 'metadata.json'`) so we can change it easily if we want later.
```suggestion
  console.log(`Writing metadata to ${METADATA_FILENAME}`)
```
etc.